modules/apps/commerce/commerce-product-type-grouped-web/src/main/resources/META-INF/resources/view_cp_definition_grouped_entries.jsp

```diff
@@ -210,39 +214,43 @@ renderResponse.setTitle(cpDefinition.getName(themeDisplay.getLanguageId()));
 	}
 </aui:script>
 
-<aui:script use="liferay-item-selector-dialog">
-	window.document
-		.querySelector('#<portlet:namespace />addDefinitionGroupedEntry')
-		.addEventListener('click', (event) => {
-			event.preventDefault();
-
-			var itemSelectorDialog = new A.LiferayItemSelectorDialog({
-				eventName: 'productDefinitionsSelectItem',
-				on: {
-					selectedItemChange: function (event) {
-						var <portlet:namespace />addCPDefinitionIds = [];
+<aui:script sandbox="<%= true %>">
+	const addDefinitionGroupedEntryButton = document.getElementById(
+		'<portlet:namespace />addDefinitionGroupedEntry'
+	);
 
-						var selectedItems = event.newVal;
-
-						if (selectedItems) {
-							window.document.querySelector(
-								'#<portlet:namespace />entryCPDefinitionIds'
-							).value = selectedItems;
+	if (addDefinitionGroupedEntryButton) {
+		addDefinitionGroupedEntryButton.addEventListener('click', (event) => {
+			event.preventDefault();
 
-							var addCPDefinitionGroupedEntryFm = window.document.querySelector(
-								'#<portlet:namespace />addCPDefinitionGroupedEntryFm'
+			Liferay.Util.openSelectionModal({
+				multiple: true,
+				onSelect: function (selectedItems) {
+					if (selectedItems && selectedItems.length) {
+						const entryCPDefinitionIds = document.getElementById(
+							'<portlet:namespace />entryCPDefinitionIds'
+						);
+
+						if (entryCPDefinitionIds) {
+							entryCPDefinitionIds.value = selectedItems.map(
+								(item) => item.value
 							);
+						}
+
+						const addCPDefinitionGroupedEntryFm = document.getElementById(
+							'<portlet:namespace />addCPDefinitionGroupedEntryFm'
+						);
 
+						if (addCPDefinitionGroupedEntryFm) {
```

Comment:
@kresimir-coko this looks overly defensive. `addCPDefinitionGroupedEntryFm` is not conditionally rendered so, when undefined, throwing an error should be more correct (it also gives more hints for fixing).

(It's absolute not a major thing, just a personal thought so feel free not to update it)

Reply:
We shouldn't delay this any further I think, so I won't change it, for now 👍

Reply:
Totally ok ;)